src/gtc/passes/gtir_definitive_assignment_analysis.py

```diff
@@ -0,0 +1,63 @@
+import copy
+import warnings
+from typing import List, Set
+
+from eve import NodeVisitor
+from gtc import gtir
+
+
+class DefinitiveAssignmentAnalysis(NodeVisitor):
+    def visit_IfStmt(self, node: gtir.FieldIfStmt, *, alive_vars: Set[str], **kwargs):
+        true_branch_vars = copy.copy(alive_vars)
+        false_branch_vars = copy.copy(alive_vars)
+        self.visit(node.true_branch, alive_vars=true_branch_vars, **kwargs)
+        self.visit(node.false_branch, alive_vars=false_branch_vars, **kwargs)
```

Comment:
We can just create new set instances without using the `copy` module
```suggestion
        self.visit(node.true_branch, alive_vars={*alive_vars}, **kwargs)
        self.visit(node.false_branch, alive_vars={*alive_vars}, **kwargs)
```